app/main.py

```diff
@@ -1,8 +1,39 @@
 class Person:
-    # write your code here
-    pass
+
+    people = {}
+
+    def __init__(self, name: "str", age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people.update({self.name: self})
+
+    def add_partner(self, partner_name: str) -> None:
+        self.partner = Person.people[partner_name]
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+    people_list = []
+    for person in people:
+        new_obj = Person(person["name"], person["age"])
+        try:
+            if person["wife"] is not None:
+                new_obj.wife = person["wife"]
+        except (KeyError, AttributeError):
+            if person["husband"] is not None:
+                new_obj.husband = person["husband"]
```

Comment:
The current logic assumes that a person can have either a 'wife' or a 'husband' attribute. However, if both keys are missing, this will raise a `KeyError`. Consider using `get` method to safely access dictionary keys without raising exceptions.